src/containers/single-event/SingleEvent.tsx

```diff
@@ -0,0 +1,77 @@
+import { Spin } from 'antd';
+import React, { useEffect } from 'react';
+import { Helmet } from 'react-helmet';
+import { connect, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import styled from 'styled-components';
+import EventDetails from '../../components/event-details/EventDetails';
+import { C4CState } from '../../store';
+import { AsyncRequest, AsyncRequestKinds } from '../../utils/asyncRequest';
+import { getUpcomingEvents } from '../upcoming-events/ducks/thunks';
+import { EventProps, EventsReducerState } from '../upcoming-events/ducks/types';
+
+const ContentContainer = styled.div`
+  padding: 24px;
+  margin: auto;
+  max-width: 1200px;
+  width: 80%;
+`;
+
+const CenteredContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+interface UpcomingEventsProps {
+  readonly events: EventsReducerState['upcomingEvents'];
+}
+
+const SingleEvent: React.FC<UpcomingEventsProps> = ({ events }) => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (events.kind === AsyncRequestKinds.NotStarted) {
```

Comment:
There are utility functions which return booleans. Try isAsyncRequestNotStarted() (I think that's what it's called?).

What if the request is failed?

Reply:
We account for that in the JSX but good catch! We'll use the method.